carsus/io/vald/vald.py

```diff
@@ -204,62 +238,60 @@ def check_wavelength_column_medium_and_units(self):
             )
         return wave_col_name, wave_air, wave_units
 
-    def parse_vald(self, vald_raw=None, strip_molecules=True):
+    def parse_vald(self, vald_raw=None):
         """
         Parses raw vald DataFrame
 
 
         Parameters
         ----------
         vald_raw: pandas.DataFrame
-        strip_molecules: bool
-            If True, remove molecules from vald
+
 
         Returns
         -------
             pandas.DataFrame
+                vald atoms
+            pandas.DataFrame
+                vald molecules
         """
 
         vald = vald_raw if vald_raw is not None else self.vald_raw.copy()
 
-        vald["elm_ion"] = vald["elm_ion"].str.replace("'", "")
+        vald.loc[:, "elm_ion"] = vald["elm_ion"].str.replace("'", "")
         vald[["chemical", "ion_charge"]] = vald["elm_ion"].str.split(" ", expand=True)
-        vald["ion_charge"] = vald["ion_charge"].astype(int) - 1
+        vald.loc[:, "ion_charge"] = vald["ion_charge"].astype(int) - 1
 
         # Check units and medium of wavelength column and create wavelength column in angstroms in vacuum
         (
             wave_col_name,
             wave_air,
             wave_units,
         ) = self.check_wavelength_column_medium_and_units()
-        vald["wavelength"] = (vald[wave_col_name].values * wave_units).to(u.AA).value
+        vald.loc[:, "wavelength"] = (
+            (vald[wave_col_name].values * wave_units).to(u.AA).value
+        )
         if wave_air:
-            vald["wavelength"] = convert_wavelength_air2vacuum(
+            vald.loc[:, "wavelength"] = convert_wavelength_air2vacuum(
                 (vald["wavelength"].values * wave_units).to(u.AA).value
             )
 
         del vald["elm_ion"]
 
-        if strip_molecules:
-            vald = self._strip_molecules(vald)
-            vald.reset_index(drop=True, inplace=True)
+        vald_atoms = vald[vald.chemical.isin(ATOMIC_SYMBOLS_DATA["symbol"])].copy()
+        vald_molecules = vald[~vald.chemical.isin(ATOMIC_SYMBOLS_DATA["symbol"])].copy()
```

Comment:
I assume there isn't a risk of weird overlap between some molecule and some element?

Reply:
I don't believe so, because of the capitalization in the atomic symbols. So this should be able to tell the difference between Hafnium Hf and Hydrogen-Fluoride HF.